rdmd.d

```diff
@@ -437,70 +393,182 @@ private void unlockWorkPath()
     }
 }
 
-// Rebuild the executable fullExe starting from modules in myDeps
-// passing the compiler flags compilerFlags. Generates one large
-// object file.
+// Internal information needed for a build
+struct BuildInformation {
+    string root;
+    string exeBasename;
+    string outExt;
+    string workDir;
+    string objDir;
+    string[string] myDeps;
+    string[] compilerFlags;
+    bool addStubMain;
+    bool makeOnlyDependencies;
+}
 
-private int rebuild(string root, string fullExe,
-        string workDir, string objDir, in string[string] myDeps,
-        string[] compilerFlags, bool addStubMain)
+/*
+Find the path to the build witness file
+If the exe file is stored in a different directory, a special witness file is
+used
+TODO: we now always store the object file in the temporary directory
+*/
+string buildWitnessPath(in ref BuildInformation build)
 {
-    version (Windows)
-        fullExe = fullExe.defaultExtension(".exe");
+    string path;
+
+    if (exe.ptr)
```

Comment:
That has different semantics. If the intention is to compare against `null`, `!is null` would be a more readable replacement.